frontend/app/routes/_gcweb-app.personal-information.preferred-language.tsx

```diff
@@ -0,0 +1,39 @@
+import { type LoaderFunctionArgs, json } from '@remix-run/node';
+import { useLoaderData } from '@remix-run/react';
+
+import { useTranslation } from 'react-i18next';
+
+import { userService } from '~/services/user-service.server';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('common', 'personal-information');
+
+export const handle = {
+  breadcrumbs: [{ labelI18nKey: 'common:personal-information.breadcrumbs.home', to: '/' }, { labelI18nKey: 'common:personal-information.page-title', to: '/personal-information' }, { labelI18nKey: 'common:preferred-language.page-title' }],
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-0004',
+  pageTitleI18nKey: 'common:preferred-language.page-title',
+} as const satisfies RouteHandleData;
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const userId = await userService.getUserId();
+  const userInfo = await userService.getUserInfo(userId);
+  return json({ user: userInfo });
```

Comment:
Should throw `404` response if `userInfo` is not found.

Reply:
Not wrong, but the entire application will require user data.. if we don't have it, we should probably catch that and handle it in the root route by maybe displaying a "you are not eligible to use this service" page.

Reply:
Yup but we are not there yet. Everything could/would go to garbage at any time. 🤷‍♂️

Reply:
@Fbasham remove this optional chaining (?.) since user shouldn't be `null` or `undefined`. :)

```tsx
<dd>{user?.preferredLanguage}</dd>
```

Reply:
kk, added.  Anything else?